vllm/model_executor/layers/quantization/compressed_tensors/schemes/compressed_tensors_24.py

```diff
@@ -254,9 +258,10 @@ def apply_weights(
             bt_meta=layer.meta,
             scale_a=input_scale,
             scale_b=layer.weight_scale,
-            out_dtype=self.output_dtype,
+            out_dtype=x.dtype,
```

Comment:
We should run some of the cases for each of tests here onwards: https://github.com/neuralmagic/vllm/blob/4c0d93f4b2de241336f4732cb5799cee8fedcb52/tests/quantization/test_compressed_tensors.py#L305

Reply:
Good call. I ran this locally on an H100 and it passes:
`pytest tests/quantization/test_compressed_tensors.py`